py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,760 @@
+    All:
+        {icon} The icon associated with a formatting section
+    format_clouds:
+        {coverage} Cloud coverage percentage
+    format_humidity:
+        {humidity} Humidity percentage
+    format_pressure:
+        {pressure} Current atmospheric pressure in Pascals
+        {sea_level} Sea-level atmospheric pressure in Pascals.
+    format_rain:
+        {amount} Rainfall in the specified unit
+        {unit} The unit specified
+    format_snow:
+        {amount} Snowfall in the specified unit
+        {unit} The unit specified
+    format_temperature:
+        {current} Current temperature
+        {max} Maximum temperature in the configured unit
+        {min} Minimum temperature
+        {unit} The unit specified
+    format_wind:
+        {degree} Current wind heading
+        {gust} Wind gusts speed in the specified unit
+        {speed} Wind speed
+        {unit} The unit specified
+    format only:
+        {city} The name of the city where the weather is
+        {forecast} Output of format_forecast
+    format, format_forecast:
+        {clouds} Output of format_clouds
+        {description} Natural description of the current weather
+        {humidity} Output of format_humidity
+        {main} Short description of the current weather
+        {pressure} Output of format_pressure
+        {snow} Output of format_snow
+        {sunrise} Output of format_sunrise
+        {sunset} Output of format_sunset
+        {temperature} Output of format_temperature
+        {wind} Output of format_wind
+
+Example configuration:
+```
+weather_owm {
+  api_key = '...'
+
+  icons {
+    200: "☔"
+    230_232: "🌧"
+  }
+}
+```
+Outputs: 🌫 ○ 59°F, foggy ⛅ ☼ 🌧`
+- Currently foggy, 59°F outside, with forecast of cloudy tomorrow, sunny the
+  next day, then rainy
+
+
+@author alexoneill
+@licence MIT
+'''
+
+import datetime
+
+
+# API information
+OWM_API = '2.5'
+OWM_CURR_ENDPOINT = 'http://api.openweathermap.org/data/%s/weather?' \
+    'APPID=%s&lat=%f&lon=%f&lang=%s'
+OWM_FUTURE_ENDPOINT = 'http://api.openweathermap.org/data/%s/forecast?' \
+    'APPID=%s&lat=%f&lon=%f&lang=%s&cnt=%%d'
+IP_ENDPOINT = 'https://timezoneapi.io/api/ip'
+
+# Paths of information to extract from JSON
+IP_LOC = '//data/location'
+IP_GMT_OFF = '//data/datetime/offset_gmt'
+OWM_CLOUD_COVER = '//clouds/all'
+OWM_DESC = '//weather:0/main'
+OWM_DESC_LONG = '//weather:0/description'
+OWM_HUMIDITY = '//main/humidity'
+OWM_PRESSURE = '//main'
+OWM_RAIN = '//rain/3h'
+OWM_SNOW = '//snow/3h'
+OWM_SUNRISE = '//sys/sunrise'
+OWM_SUNSET = '//sys/sunset'
+OWM_TEMP = '//main'
+OWM_WEATHER_ICON = '//weather:0/id'
+OWM_WIND = '//wind'
+
+# Units constants
+RAIN_UNITS = set(['cm', 'ft', 'in', 'mm', 'm', 'yd'])
+SNOW_UNITS = RAIN_UNITS
+TEMP_UNITS = set(['c', 'f', 'k'])
+WIND_UNITS = set(['fsec', 'msec', 'mph', 'kmh'])
+
+# Conversion factors
+FT_FROM_METER = 3.28084
+IN_FROM_MM = 0.0393701
+KMH_FROM_MSEC = 0.277778
+MPH_FROM_MSEC = 2.23694
+
+# Thresholds defaults
+THRESHOLDS = [
```

Comment:
Would you like this thresholds? I think it's a bit more... **_COLORFUL_**! 🌈 
```
        THRESHOLDS = [
            (-65, '#111111'),
            (-60, '#21006B'),
            (-55, '#4C006B'),
            (-50, '#6B006B'),
            (-45, '#820082'),
            (-40, '#990099'),
            (-35, '#B300B3'),
            (-30, '#CC00CC'),
            (-25, '#E600E6'),
            (-20, '#FF00FF'),
            (-15, '#D100FF'),
            (-10, '#9E00FF'),
            (-5, '#6600FF'),
            (0, '#0000FF'),
            (5, '#004AFF'),
            (10, '#0073FF'),
            (15, '#00A3FF'),
            (20, '#00CCFF'),
            (25, '#00E6FF'),
            (30, '#00FFFF'),
            (35, '#00FFB3'),
            (40, '#7FFF00'),
            (45, '#CEFF00'),
            (50, '#FFFF00'),
            (55, '#FFE600'),
            (60, '#FFCC00'),
            (65, '#FFAE00'),
            (70, '#FF9900'),
            (75, '#FF7F00'),
            (80, '#FF4F00'),
            (85, '#FF0000'),
            (90, '#FF4545'),
            (95, '#FF6868'),
            (100, '#FF8787'),
            (105, '#FF9E9E'),
            (110, '#FFB5B5'),
            (115, '#FFCFCF'),
            (120, '#FFE8E8'),
            (125, '#EEEEEE')
        ]
````